api_tests/tests/sanity.ts

```diff
@@ -152,4 +160,31 @@ describe("Sanity", () => {
             ).rejects.toMatchObject(expectedProblem);
         })
     );
+
+    it(
+        "bob-connects-to-alice-using-config",
+        createAliceAndBob(async ([alice, bob]) => {
+            await alice.cndInstance.start();
+
+            const aliceId = await alice.cnd.getPeerId();
+
+            const aliceAddresses = await alice.cnd.getPeerListenAddresses();
+
+            const configOverride = {
+                network: {
+                    peer_addresses: aliceAddresses,
+                },
+            };
+
+            const currentConfig = bob.cndInstance.getConfigFile();
+
+            const updatedConfig = merge(configOverride, currentConfig);
```

Comment:
This only works if the currentConfig doesn't already specify the `peer_addresses` field. I'd flip it so you an be sure that your overrides always apply.
```suggestion
            const updatedConfig = merge(currentConfig, configOverride);
```

Reply:
well spotted